modules/data.atmosphere/R/download.ICOS.R

```diff
@@ -0,0 +1,237 @@
+          strptime(response$results$bindings$timeEnd$value, format = "%Y-%m-%dT%H:%M:%S")
+        )
+      if (is.null(dataset_url)) {
+        PEcAn.logger::logger.severe("Data is not available for the requested site")
+      }
+      if (dataset_start_date > lubridate::as_datetime(start_date)) {
+        PEcAn.logger::logger.severe(
+          paste(
+            "Data is not available for the requested start date. Please try again with",
+            dataset_start_date,
+            "as start date."
+          )
+        )
+      }
+      
+      if (dataset_end_date < lubridate::as_date(end_date)) {
+        PEcAn.logger::logger.severe(
+          paste(
+            "Data is not available for the requested end date. Please try again with",
+            dataset_end_date,
+            "as end date."
+          )
+        )
+      }
+      dataset_id <- sub(".*/", "", dataset_url)
+      
+      # construct the download URL
+      download_url <-
+        paste0('https://data.icos-cp.eu/licence_accept?ids=%5B%22',
+               dataset_id,
+               '%22%5D')
+      # Download the zip file
+      file <-
+        httr::GET(url = download_url,
+                  httr::write_disk(zip_file_name,
+                                   overwrite = TRUE),
+                  httr::progress())
+    }
+    
+    if (extract_file_flag) {
+      # extract only the hourly data file
+      zipped_csv_name <-
+        grep(
+          paste0('*', file_name),
+          utils::unzip(zip_file_name, list = TRUE)$Name,
+          ignore.case = TRUE,
+          value = TRUE
+        )
+      utils::unzip(zip_file_name,
+                   files = zipped_csv_name,
+                   junkpaths = TRUE,
+                   exdir = outfolder)
+      if (tolower(product) == "drought2018") {
+        output_file_name <- zipped_csv_name
+      }
+    }
+    
+    
+    # get start and end year of data from file
+    firstline <-
+      system(paste0("head -2 ", file.path(outfolder, output_file_name)), intern = TRUE)
+    firstline <- firstline[2]
+    lastline <-
+      system(paste0("tail -1 ", file.path(outfolder, output_file_name)), intern = TRUE)
+    
+    firstdate_st <- paste0(
+      substr(firstline, 1, 4),
+      "-",
+      substr(firstline, 5, 6),
+      "-",
+      substr(firstline, 7, 8),
+      " ",
+      substr(firstline, 9, 10),
+      ":",
+      substr(firstline, 11, 12)
+    )
+    lastdate_st <- paste0(
+      substr(lastline, 1, 4),
+      "-",
+      substr(lastline, 5, 6),
+      "-",
+      substr(lastline, 7, 8),
+      " ",
+      substr(lastline, 9, 10),
+      ":",
+      substr(lastline, 11, 12)
+    )
+    
+    
+    rows    <- 1
+    results <- data.frame(
+      file = character(rows),
+      host = character(rows),
+      mimetype = character(rows),
+      formatname = character(rows),
+      startdate = character(rows),
+      enddate = character(rows),
+      dbfile.name = basename(output_file_name),
```

Comment:
this one needs to be without the file extension I think
```suggestion
      dbfile.name = substr(basename(output_file_name), 1, nchar(basename(output_file_name)) - 4),
```